x-pack/plugins/case/server/client/cases/update.ts

```diff
@@ -95,133 +202,158 @@ export const update = ({
-          savedObject: {
-            ...myCase,
-            ...updatedCase,
-            attributes: { ...myCase.attributes, ...updatedCase?.attributes },
-            references: myCase.references,
-            version: updatedCase?.version ?? myCase.version,
-          },
-        });
-      });
+  // If syncAlerts setting turned on we need to update all alerts' status
+  // attached to the case to the current status.
+  const casesWithSyncSettingChangedToOn = updateFilterCases.filter((caseToUpdate) => {
+    const currentCase = myCases.saved_objects.find((c) => c.id === caseToUpdate.id);
+    return (
+      currentCase != null &&
+      caseToUpdate.settings?.syncAlerts != null &&
+      currentCase.attributes.settings.syncAlerts !== caseToUpdate.settings.syncAlerts &&
+      caseToUpdate.settings.syncAlerts
+    );
+  });
 
-    await userActionService.postUserActions({
+  for (const theCase of [...casesWithSyncSettingChangedToOn, ...casesWithStatusChangedAndSynced]) {
+    const currentCase = myCases.saved_objects.find((c) => c.id === theCase.id);
+    const totalComments = await caseService.getAllCaseComments({
       client: savedObjectsClient,
-      actions: buildCaseUserActions({
-        originalCases: myCases.saved_objects,
-        updatedCases: updatedCases.saved_objects,
-        actionDate: updatedDt,
-        actionBy: { email, full_name, username },
-      }),
+      id: theCase.id,
+      options: {
+        fields: [],
+        filter: `${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.alert} OR ${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.generatedAlert}`,
+        page: 1,
+        perPage: 1,
+      },
     });
 
-    return CasesResponseRt.encode(returnUpdatedCase);
+    // TODO: double check that this logic will get all sub case comments and include them in the updates
+    const caseComments = (await caseService.getAllCaseComments({
+      client: savedObjectsClient,
+      id: theCase.id,
+      includeSubCaseComments: true,
+      options: {
+        fields: [],
+        filter: `${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.alert} OR ${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${CommentType.generatedAlert}`,
+        page: 1,
+        perPage: totalComments.total,
+      },
+      // The filter guarantees that the comments will be of type alert
+    })) as SavedObjectsFindResponse<{ alertId: string | string[]; index: string }>;
+
+    // TODO: comment about why we need this (aka alerts might come from different indices? so dedup them)
+    const idsAndIndices = caseComments.saved_objects.reduce(
+      (acc: { ids: string[]; indices: Set<string> }, comment) => {
+        const alertId = comment.attributes.alertId;
+        const ids = Array.isArray(alertId) ? alertId : [alertId];
+        acc.ids.push(...ids);
+        acc.indices.add(comment.attributes.index);
+        return acc;
+      },
+      { ids: [], indices: new Set<string>() }
```

Comment:
I had a feeling that we gonna need that functionality a lot. If that's the case maybe a helper function would be nice.

Reply:
good point, I'll move this to a helper 👍